### PR TITLE
[CALCITE-5688] Support TRUNCATE TABLE DDL statement in Babel parser

### PR DESCRIPTION
**Pull request by sumeetgajjar**
The PR adds support for TRUNCATE TABLE statement to the babel parser.

### CODE REVIEW

**Comment by sumeetgajjar** [comment]
@JiajunBernoulli @julianhyde can you please review this PR?

**Comment by JiajunBernoulli** [comment]
Run `./gradlew autostyleApply` to fix the violations.

More information on `gradlew` for your reference:
https://calcite.apache.org/docs/howto.html#building-from-a-source-distribution

**Comment by sumeetgajjar** [comment]
> Run `./gradlew autostyleApply` to fix the violations.
> 
> More information on `gradlew` for your reference: https://calcite.apache.org/docs/howto.html#building-from-a-source-distribution

Hi @JiajunBernoulli - thank you for the pointer. Fixed the style violations.

I wasn't aware about these since this is my first time contributing to calcite.
I shall keep this in mind for my upcoming contributions.

**Review by JiajunBernoulli — APPROVED** [review]

File: core/src/main/codegen/templates/Parser.jj [hunk 1]
```diff
@@ -1136,6 +1137,8 @@ SqlNode SqlStmt() :
         stmt = SqlDrop()
     |
 </#if>
+        stmt = SqlTruncateTable()
```
Comment (JiajunBernoulli) [inline core/src/main/codegen/templates/Parser.jj:1140]: Please implement it in Babel.

**Review by JiajunBernoulli — CHANGES_REQUESTED** [review]

**Comment by JiajunBernoulli** [comment]
@sumeetgajjar 
Do you still have time to move it to Babel? If not, I will move it. 
And save your name in the commit.

**Comment by julianhyde** [comment]
@JiajunBernoulli Thanks for being proactive in getting PRs completed and merged. I often do the same. If you make the significant changes, you should consider adding yourself as co-author (add two `Co-authored-by:` lines in the commit message).

**Pull request closed without merging**